File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    var answer = 0;
    
    let burgur = [];
    
    [...ingredient].forEach((ing) => {
        if(burgur[burgur.length-1] === 1 && ing === 2)
            burgur[burgur.length-1] = 12
        else if(burgur[burgur.length-1] === 12 && ing === 3)
            burgur[burgur.length-1] = 123
        else if(burgur[burgur.length-1] === 123 && ing === 1){
            burgur.pop();
            answer++;
        }
        else
            burgur.push(ing);       
    })
    
    return answer;
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> ingredient) {
    int answer = 0;
    
    vector<int> burgur; 
    for(auto ing : ingredient){
        if(burgur.empty()){
            burgur.push_back(ing);
            continue;
        }
        
        if(burgur.back() == 1 && ing == 2)
            burgur.back() = 12;
        else if(burgur.back() == 12 && ing == 3)
            burgur.back() = 123;
        else if(burgur.back() == 123 && ing == 1){
            burgur.pop_back();
            answer++;
        }
        else
            burgur.push_back(ing);
    }
    
    return answer;
}